### PR TITLE
[runtime] Move 'pedump' to tools/pedump, build tools by default.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

**Comment by lewurm** [comment]
```**Error**: automake failed.
scripts/ci/babysitter: Test suite terminated with code 1, and suite cannot report test case data. Halting.
*** end(11): configure: Failed```
something is off.

File: Makefile.am [hunk 1]
```diff
@@ -7,12 +7,10 @@ MONOTOUCH_SUBDIRS = $(libgc_dir) eglib/src mono
 if CROSS_COMPILING
 SUBDIRS = po $(libgc_dir) eglib mono $(ikvm_native_dir) support data runtime scripts man samples msvc $(docs_dir) acceptance-tests llvm
 # Keep in sync with SUBDIRS
-## 'tools' is not normally built
 DIST_SUBDIRS = m4 po $(libgc_dir) eglib mono ikvm-native support data runtime scripts man samples tools msvc docs acceptance-tests llvm
 else
-SUBDIRS = po $(libgc_dir) eglib mono $(ikvm_native_dir) support data runtime scripts man samples msvc $(docs_dir) acceptance-tests llvm
+SUBDIRS = po $(libgc_dir) eglib mono $(ikvm_native_dir) support data runtime scripts man samples tools msvc $(docs_dir) acceptance-tests llvm
```
Comment (akoeplinger) [inline Makefile.am]: add the same in the CROSS_COMPILING section above too to make it consistent.

Actually, I'm wondering if we still need this separation when both if branches are the same anyway?

**Comment by akoeplinger** [comment]
seems monograph broke while nobody was building it (which is why I like building tools by default 👍 ):

```
monograph.o: In function `method_stats':
/mnt/jenkins/workspace/test-mono-pull-request-i386/tools/monograph/monograph.c:240: undefined reference to `mono_opcodes'
monograph.o: In function `print_method':
/mnt/jenkins/workspace/test-mono-pull-request-i386/tools/monograph/monograph.c:627: undefined reference to `mono_opcodes'
monograph.o: In function `mono_method_find_bblocks':
/mnt/jenkins/workspace/test-mono-pull-request-i386/tools/monograph/monograph.c:801: undefined reference to `mono_opcodes'
collect2: error: ld returned 1 exit status
make[3]: *** [monograph] Error 1
```

**Comment by alexrp** [comment]
monograph is showing some undefined references to `mono_read32`.

**Comment by lewurm** [comment]
looks good now?

**Comment by vargaz** [comment]
Everybody ok with this ?

**Comment by akoeplinger** [comment]
@vargaz please fix https://github.com/mono/mono/pull/3133/files#r66353757, otherwise I'm 👍

**Comment by alexrp** [comment]
LGTM

**Comment by lewurm** [comment]
some of the tools fail when being cross compiled, e.g. on `mono-master-monodroid`:

```[2:42:37] 4 warnings generated.
[2:42:37]   AR       libsgen-grep-binprot32p.a
[2:42:37]   CCLD     sgen-grep-binprot
[2:42:37] /Users/builder/data/lanes/1845/8b57a819/source/mono/mono/sgen/sgen-protocol-def.h:32: error: undefined reference to '__aeabi_l2d'
[2:42:37] /Users/builder/data/lanes/1845/8b57a819/source/mono/mono/sgen/sgen-protocol-def.h:32: error: undefined reference to '__aeabi_dmul'
[2:42:37] /Users/builder/data/lanes/1845/8b57a819/source/mono/mono/sgen/sgen-protocol-def.h:32: error: undefined reference to '__aeabi_l2d'
[2:42:37] /Users/builder/data/lanes/1845/8b57a819/source/mono/mono/sgen/sgen-protocol-def.h:32: error: undefined reference to '__aeabi_ddiv'
[2:42:37] /Users/builder/data/lanes/1845/8b57a819/source/mono/mono/sgen/sgen-protocol-def.h:81: error: undefined reference to '__aeabi_l2d'
[2:42:37] /Users/builder/data/lanes/1845/8b57a819/source/mono/mono/sgen/sgen-protocol-def.h:81: error: undefined reference to '__aeabi_dmul'
[2:42:37] /Users/builder/data/lanes/1845/8b57a819/source/mono/mono/sgen/sgen-protocol-def.h:81: error: undefined reference to '__aeabi_l2d'
[2:42:37] /Users/builder/data/lanes/1845/8b57a819/source/mono/mono/sgen/sgen-protocol-def.h:81: error: undefined reference to '__aeabi_ddiv'
[2:42:37] /Users/builder/data/lanes/1845/8b57a819/source/mono/mono/sgen/sgen-protocol-def.h:32: error: undefined reference to '__aeabi_dmul'
[2:42:37] /Users/builder/data/lanes/1845/8b57a819/source/mono/mono/sgen/sgen-protocol-def.h:32: error: undefined reference to '__aeabi_ddiv'
[2:42:37] /Users/builder/data/lanes/1845/8b57a819/source/mono/mono/sgen/sgen-protocol-def.h:81: error: undefined reference to '__aeabi_dmul'
[2:42:37] /Users/builder/data/lanes/1845/8b57a819/source/mono/mono/sgen/sgen-protocol-def.h:81: error: undefined reference to '__aeabi_ddiv'
[2:42:37] clang38: error: linker command failed with exit code 1 (use -v to see invocation)
[2:42:37] make[6]: *** [sgen-grep-binprot] Error 1
[2:42:37] make[5]: *** [all-recursive] Error 1
[2:42:37] make[4]: *** [all-recursive] Error 1
[2:42:37] make[3]: *** [all] Error 2
[2:42:37] make[2]: *** [.stamp-build-target-armv6] Error 1
[2:42:37] make[1]: *** [mono] Error 2```

**Pull request merged**